kubernetes/resource_kubernetes_stateful_set_v1_test.go

```diff
@@ -351,6 +353,31 @@ func testAccCheckKubernetesStatefulSetV1Destroy(s *terraform.State) error {
 				return fmt.Errorf("StatefulSet still exists: %s: (Generation %#v)", rs.Primary.ID, resp.Status.ObservedGeneration)
 			}
 		}
+
+		// StatefulSet can create a PVC via volumeClaimTemplate. However, once the StatefulSet is removed, the PVC remains.
+		// There is a beta feature(persistentVolumeClaimRetentionPolicy) since 1.27 that aims to address this problem:
+		// - https://kubernetes.io/blog/2021/12/16/kubernetes-1-23-statefulset-pvc-auto-deletion/
+		// - https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#persistentvolumeclaim-retention
+		// By default, StatefulSetAutoDeletePVC feature is not enabled in the feature gate.
+		// That is why we clean up resources manually here.
+		pvc, err := conn.CoreV1().PersistentVolumeClaims(namespace).List(ctx, metav1.ListOptions{})
```

Comment:
You can filter the returned resources by name directly here, and not need to do the `for` loop below anymore.
Just pass `metav1.ListOptions{FieldSelector="metadata.name="+name}` instead of empty.

Reply:
As discussed internally, skip this for now, since `FieldSelector` requires an exact name match and `StatefulSet` adds an index postfix to a Pod and thus PVC name.

It is a good proposal! Thanks.